pkg/workloads/console/runner/runner.go

```diff
@@ -0,0 +1,162 @@
+package runner
```

Comment:
I'm not a big fan of this name. If anyone has any better suggestions then please shout!

Reply:
I can see where you're coming from here, but I think that moving that method into anu would have a couple of downsides:
- It's going to be a lot harder/messier to test - I think that the tests are quitely nicely wrapped up at the moment
- Anyone else wanting to create consoles is going to have to duplicate the logic we've encoded in anu (while this is minimal at the moment, I expect that this method is going to grow a bit)

I did try and think about this for a while and played with the code in different places (after starting out with it all in anu) and this is the split that I disliked the least. It still doesn't feel totally natural though!

cc @lawrencejones, who probably has a more eloquent opinion on this

Reply:
I'm confused - there's basically no logic in `Create`, right? We're just generating a name and then creating the console. What is there to test?

Reply:
If we really want to leave `Create` in here, then it might be better as a convenience constructor, either in `types` or in a `consoles` package.

Reply:
> I'm confused - there's basically no logic in Create, right? We're just generating a name and then creating the console. What is there to test?

And also setting up the object reference to the template. And in the future copying over (some of?) the labels from the template to the console itself. We may also want to enforce that the console is created in the same namespace that the template exists in? It feels to me that it's a bit too early to tell how this'll play out.

Given that we're not doing any testing in anu, it feels nicer to be able to call an external interface such as this and have some confidence that it's not going to explode. But I accept that that argument doesn't hold too much weight in this case.

Reply:
Fair, I buy that. Happy to go with this and we'll see if it grows into a horrible monster in time.

Reply:
(really not a fan of the package name though. Could be called a processor or a broker or a manager or a handler or a supervisor and I'd just as much of a clue what it's meant to do)

Reply:
> Fair, I buy that. Happy to go with this and we'll see if it grows into a horrible monster in time.

Yeah I fully expect to land on the wrong side of history with this one! 😆